src/gui/core/canvas_private.hpp

```diff
@@ -446,6 +447,9 @@ class text_shape : public canvas::shape {
 
 	/** The maximum height for the text. */
 	typed_formula<int> maximum_height_;
+
+	/** Whether to draw a border around the text. */
+	typed_formula<bool> outline_;
```

Comment:
I think it may make more sense for this to be an integer (representing the thickness of the outline)? I'm not sure _how_ useful it would be though…